src/transforms/stack.js

```diff
@@ -112,51 +114,59 @@ function extent(stack, Y2) {
   return [min, max];
 }
 
-function offsetExpand(stacks, Y1, Y2) {
-  for (const stack of stacks) {
-    const [yn, yp] = extent(stack, Y2);
-    for (const i of stack) {
-      const m = 1 / (yp - yn || 1);
-      Y1[i] = m * (Y1[i] - yn);
-      Y2[i] = m * (Y2[i] - yn);
+function offsetExpand(allstacks, Y1, Y2) {
+  for (const stacks of allstacks) {
+    for (const stack of stacks) {
+      const [yn, yp] = extent(stack, Y2);
+      for (const i of stack) {
+        const m = 1 / (yp - yn || 1);
+        Y1[i] = m * (Y1[i] - yn);
+        Y2[i] = m * (Y2[i] - yn);
+      }
     }
-  }
+  } 
 }
 
-function offsetSilhouette(stacks, Y1, Y2) {
-  for (const stack of stacks) {
-    const [yn, yp] = extent(stack, Y2);
-    for (const i of stack) {
-      const m = (yp + yn) / 2;
-      Y1[i] -= m;
-      Y2[i] -= m;
+function offsetSilhouette(allstacks, Y1, Y2) {
+  for (const stacks of allstacks) {
+    for (const stack of stacks) {
+      const [yn, yp] = extent(stack, Y2);
+      for (const i of stack) {
+        const m = (yp + yn) / 2;
+        Y1[i] -= m;
+        Y2[i] -= m;
+      }
     }
+    offsetZero(stacks, Y1, Y2);
   }
-  offsetZero(stacks, Y1, Y2);
+  offsetCenterFacets(Y1, Y2, allstacks.map(d => d.flat()));
 }
 
-function offsetWiggle(stacks, Y1, Y2, Z) {
-  const prev = new InternMap();
-  let y = 0;
-  for (const stack of stacks) {
-    let j = -1;
-    const Fi = stack.map(i => Math.abs(Y2[i] - Y1[i]));
-    const Df = stack.map(i => {
-      j = Z ? Z[i] : ++j;
-      const value = Y2[i] - Y1[i];
-      const diff = prev.has(j) ? value - prev.get(j) : 0;
-      prev.set(j, value);
-      return diff;
-    });
-    const Cf1 = [0, ...cumsum(Df)];
-    for (const i of stack) {
-      Y1[i] += y;
-      Y2[i] += y;
+function offsetWiggle(allstacks, Y1, Y2, Z) {
+  for (const stacks of allstacks) {
+    const prev = new InternMap();
+    let y = 0;
+    for (const stack of stacks) {
+      let j = -1;
+      const Fi = stack.map(i => Math.abs(Y2[i] - Y1[i]));
+      const Df = stack.map(i => {
+        j = Z ? Z[i] : ++j;
+        const value = Y2[i] - Y1[i];
+        const diff = prev.has(j) ? value - prev.get(j) : 0;
+        prev.set(j, value);
+        return diff;
+      });
+      const Cf1 = [0, ...cumsum(Df)];
+      for (const i of stack) {
+        Y1[i] += y;
+        Y2[i] += y;
+      }
+      const s1 = sum(Fi);
+      if (s1) y -= sum(Fi, (d, i) => (Df[i] / 2 + Cf1[i]) * d) / s1;
     }
-    const s1 = sum(Fi);
-    if (s1) y -= sum(Fi, (d, i) => (Df[i] / 2 + Cf1[i]) * d) / s1;
+    offsetZero(stacks, Y1, Y2);
   }
-  offsetZero(stacks, Y1, Y2);
+  offsetCenterFacets(Y1, Y2, allstacks.map(d => d.flat()));
```

Comment:
Can we avoid this flatMap by using the same signature? `offsetCenterFacets(allstacks, Y1, Y2)`